hedera-mirror-rest-java/src/main/java/com/hedera/mirror/restjava/common/Filter.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.mirror.restjava.common;
+
+import com.hedera.mirror.restjava.exception.MismatchedTypeException;
+import org.jooq.Field;
+
+public record Filter<T>(Field<?> field, RangeOperator operator, T value, Class<T> type) {
+
+    public Filter {
+        if (field.getType() != type) {
+            throw new MismatchedTypeException(field.getType(), type);
+        }
+    }
```

Comment:
Is the extra type parameter really needed? Doesn't the Field and the `T value` have the necessary type info?

```suggestion
public record Filter<T>(Field<?> field, RangeOperator operator, T value) {
```

Reply:
Yes. The other way to get the class type during runtime of reflection, of cause we dont want to pay the huge cost as this is frequently instantiated